azure/iac/sql_alert_policy.rego

```diff
@@ -1,91 +1,141 @@
 package rule
 
 #
-# PR-AZR-0103-ARM
+
 #
 
 # https://docs.microsoft.com/en-us/azure/templates/microsoft.sql/servers/securityalertpolicies
 # https://docs.microsoft.com/en-us/azure/templates/microsoft.sql/managedinstances/securityalertpolicies
 
+# PR-AZR-0102-ARM
+
 default sql_server_alert = null
 
+azure_attribute_absence["sql_server_security_alert_disabled"] {
+    resource := input.resources[_]
+    lower(resource.type) == "microsoft.sql/servers"
+    sql_resources := resource.resources[_]
+    lower(sql_resources.type) == "securityalertpolicies"
+    not sql_resources.properties.state
+}
+
+azure_attribute_absence["sql_server_security_alert_disabled"] {
+    resource := input.resources[_]
+    lower(resource.type) == "microsoft.sql/servers/securityalertpolicies"
+    not resource.properties.state
+}
+
+
 azure_sql_security_alert_disabled["sql_server_security_alert_disabled"] {
     resource := input.resources[_]
     lower(resource.type) == "microsoft.sql/servers"
     sql_resources := resource.resources[_]
     lower(sql_resources.type) == "securityalertpolicies"
-    sql_resources.properties.state == "Disabled"
+    lower(sql_resources.properties.state) == "disabled"
 }
 
 azure_sql_security_alert_disabled["sql_server_security_alert_disabled"] {
     resource := input.resources[_]
     lower(resource.type) == "microsoft.sql/servers/securityalertpolicies"
-    resource.properties.state == "Disabled"
+    lower(resource.properties.state) == "disabled"
 }
 
 sql_server_alert {
     lower(input.resources[_].type) == "microsoft.sql/servers"
+    not azure_attribute_absence["sql_server_security_alert_disabled"]
     not azure_sql_security_alert_disabled["sql_server_security_alert_disabled"]
 }
 
+sql_server_alert = false {
+    azure_attribute_absence["sql_server_security_alert_disabled"]
+}
+
 sql_server_alert = false {
     azure_sql_security_alert_disabled["sql_server_security_alert_disabled"]
 }
 
+sql_server_alert_err = "Security alert for SQL server missing in the resource!" {
+    azure_attribute_absence["sql_server_security_alert_disabled"]
+}
+
 sql_server_alert_err = "Security alert for SQL server is Disabled" {
```

Comment:
`Security alert is currently not enabled on SQL Server resource.`

Reply:
fixed